packages/beacon-node/src/network/peers/peerManager.ts

```diff
@@ -332,18 +326,24 @@ export class PeerManager {
     });
     if (peerData) {
       const oldMetadata = peerData.metadata;
+      const csc = (metadata as Partial<peerdas.Metadata>).csc ?? this.config.CUSTODY_REQUIREMENT;
+      const nodeId = peerData?.nodeId ?? computeNodeId(peer);
+      const custodySubnets = (csc !== oldMetadata?.csc || oldMetadata?.custodySubnets == null)? getDataColumnSubnets(nodeId, csc) : oldMetadata?.custodySubnets;
       peerData.metadata = {
         seqNumber: metadata.seqNumber,
         attnets: metadata.attnets,
         syncnets: (metadata as Partial<peerdas.Metadata>).syncnets ?? BitArray.fromBitLen(SYNC_COMMITTEE_SUBNET_COUNT),
         csc:
           (metadata as Partial<peerdas.Metadata>).csc ??
-          this.discovery?.["peerIdToCustodySubnetCount"].get(peer.toString()) ??
+          // TODO: from peerDAS, all metadata should have csc field
+          // may consider to disconnect them to have more qualified peers
           this.config.CUSTODY_REQUIREMENT,
+        custodySubnets,
       };
-      if (oldMetadata === null || oldMetadata.csc !== peerData.metadata.csc) {
-        void this.requestStatus(peer, this.statusCache.get());
-      }
+      // TODO: why request status again?
```

Comment:
should request metadata if metadata is old or non existent

Reply:
need to request status again in order to update network's datacolumns of this peer, will add that to the comment